### DISCUSSION TITLE
How to pass a socks proxy to Socket.io client

### DISCUSSION THREAD

**Discussion by AxelConceicao** [socketio/socket.io#4926] [Q&A] [answered]
I would like to pass a socks proxy to my client-side Socket.io instance.  
After an hour of googling I'm unabled to find atleast one tutorial about this.  

```
import { io } from 'socket.io-client'

const proxy = 'host:ip:username:password'
const socket = io('wss://example.com/')
socket.onAny((message) => {
  console.log(message)
})
```

I tried to pass a `SocksProxyAgent` to the io Manager `agent` from https://www.npmjs.com/package/socks-proxy-agent  
But io Mananer is waiting for a `string` not `SocksProxyAgent`

**Comment by darrachequesne** [comment, marked as answer]
Hi! The type of the `agent` option is indeed wrong, but fixing this would require providing different types for Node.js and browser clients: https://github.com/socketio/engine.io-client/blob/fa479164251dd1f283dd163143b546582161339a/lib/socket.ts#L40-L47

Providing the `agent` (and ignoring the typescript error) should work though.
